src/engine/app.rs

```diff
@@ -221,51 +221,57 @@ impl Application {
 }
 
 /// Helper builder for Applications.
-pub struct ApplicationBuilder<T>
+pub struct ApplicationBuilder<'a, T>
     where T: State + 'static
 {
     config: DisplayConfig,
     initial_state: T,
-    planner: Planner<()>,
+    dispatcher_builder: DispatcherBuilder<'a, 'a>,
+    world: World,
 }
 
-impl<T> ApplicationBuilder<T>
+impl<'a, T> ApplicationBuilder<'a, T>
     where T: State + 'static
 {
     /// Creates a new ApplicationBuilder with the given initial game state and
     /// display configuration.
-    pub fn new(initial_state: T, cfg: DisplayConfig) -> ApplicationBuilder<T> {
-        let pool = Arc::new(ThreadPool::new(num_cpus::get()));
+    pub fn new(initial_state: T, cfg: DisplayConfig) -> ApplicationBuilder<'a, T> {
+        let pool = Arc::new(ThreadPool::new( ::rayon::Configuration::new().num_threads(num_cpus::get())).unwrap());
```

Comment:
Could you format the code in some way? Seems there is unnecessary whitespace. And please add an import statement inside the function.

Additionally, consider using `expect` over `unwrap`.